src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySourceConfig.java

```diff
@@ -35,21 +40,35 @@
 /**
  * Holds configuration required for configuring {@link BigQuerySource}.
  */
-public final class BigQuerySourceConfig extends BigQueryBaseConfig {
+public final class BigQuerySourceConfig extends PluginConfig {
```

Comment:
Why this change? The reason BigQuery source config extends from BigQuery base config is so the same configuration class can be referenced in sources, sinks and BQ Pushdown.

Reply:
This is the major purpose for the migration, We need to extract connection config into BigQueryConnector config. The first step is to migrate BigQuery source, will migrate BigQueryTarget and probably BQ Pushdown later. I will send you some related design doc offline :)

Reply:
`BigQueryBaseConfig` itself is fine, but it inherits from `GCPConfig` which we need to decouple from because it contains connection properties.

Actually in a final state, we can have `BigQueryConnectorConfig` in `BigQueryBaseConfig` as a sub config and make `BigQueryBaseConfig` not inherit from `GCPConfig`. But for now we are only migrating BigQuerySource since `BigQueryBaseConfig` is also inherited by BigQuerySink , so in the first step , I have to decouple it.

Will need to do a refactoring when all BigQuery Source and Sink are migrated.

Reply:
This will cause a conflict with the BQ Pushdown changes in https://github.com/data-integrations/google-cloud/pull/659/files#diff-1e91c596d76d44c6146342154a5f84c569e2d639ef56eb4020251d4d6fd0a538

Reply:
no worries, you can merge first, and I will resolve the conflicts. Only one method in the BigQuerySourceUtils is impacted. That method actually just need the bucket name, no need to pass in the whole config.

Reply:
the latest commit is based on the latest develop branch. As I mentioned, BigQueryBaseConfig is fine. The only issue is it inherits from GCPConfig. We can still inherit from BigQueryBaseConfig in the future and put BigQueryConnectorConfig in it.

Reply:
discussed offline, we will leave the code as it is for now. once we migrate all BQ related source/sink to connection , we can do what I suggested above.